federatedscope/core/trainers/trainer_nbafl.py

```diff
@@ -63,8 +63,12 @@ def init_nbafl_ctx(base_trainer):
     ctx.regularizer = get_regularizer(cfg.regularizer.type)
 
     # set noise scale during upload
+    if cfg.trainer.type == 'nodefullbatch_trainer':
+        num_train_data = sum(ctx.train_loader.dataset[0]['train_mask'])
```

Comment:
Maybe we should wrap all the dataset/dataloader with a unified attribute `num_samples` in the future.

Reply:
Agree.